### ISSUE TITLE
Fast RCNN - Do the ground truths get propagated during training ?

### ISSUE THREAD

**Issue by HaziqRazali** [vlfeat/matconvnet#810]
Are the ground truths directly propagated through the network during training or are proposals the only ones being propagated i.e. the ground truth is there merely to select positive and negative proposals based on IOU ?

I ask because I am having some problems with the ground truths of my database.

**Comment by albanie** [comment]
The ground truth is only used to select positive and negative proposals.  What kind of issues are you running into?

**Comment by HaziqRazali** [comment]
Hi albanie, thanks for your continued help. I was browsing through the code and I noted this section of code under attach_proposals.m which invokes the error message if the iou of the training or validation set is lesser than 1. It thus made me wonder if the bounding boxes fed through the network during training must be perfect in terms of iou or if the ground truths themselves must be passed through the network
```

  iou = bbox_overlap(single(pbox),single(gtbb));
  [max_iou,gt_assignments] = max(iou,[],2);
  if imdb.images.set(i)<3 && all(max_iou<1)
    error('Ground truth boxes are not added!');
  end
```

**Comment by albanie** [comment]
Yes, you are quite right - good spot! The ground truth is added to the proposals [here](https://github.com/vlfeat/matconvnet/blob/master/examples/fast_rcnn/datasets/attach_proposals.m#L34-L36). Sorry - missed that on my previous look :)
